File: app/src/main/java/com/petproject/andy/bcstorage/data/BarcodeViewModel.java
package com.petproject.andy.bcstorage.data;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.petproject.andy.bcstorage.data.BarcodeRepository;
import com.petproject.andy.bcstorage.data.database.Barcode;

import java.util.List;

public class BarcodeViewModel extends AndroidViewModel {

    private BarcodeRepository mRepository;

    private LiveData<List<Barcode>> mAllBarcodes;

    public BarcodeViewModel (Application application) {
        super(application);
        mRepository = new BarcodeRepository(application);
        mAllBarcodes = mRepository.getAllBarcodes();
    }

    public LiveData<List<Barcode>> getAllBarcodes() { return mAllBarcodes; }
    public LiveData<List<Barcode>> getSpecificBarcodes(String barcode) {
        return mRepository.getSpecificBarcodes(barcode);
    }

    public void insert(Barcode barcode) { mRepository.insert(barcode); }
    public void delete(Barcode barcode) { mRepository.delete(barcode); }
    public void update(Barcode barcode) { mRepository.update(barcode); }
}

File: app/src/main/java/com/petproject/andy/bcstorage/data/BarcodeViewModel.kt
package com.petproject.andy.bcstorage.data

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData

import com.petproject.andy.bcstorage.data.BarcodeRepository
import com.petproject.andy.bcstorage.data.database.Barcode

class BarcodeViewModel(application: Application) : AndroidViewModel(application) {

    private val mRepository: BarcodeRepository

    val allBarcodes: LiveData<List<Barcode>>

    init {
        mRepository = BarcodeRepository(application)
        allBarcodes = mRepository.allBarcodes
    }

    fun getSpecificBarcodes(barcode: String): LiveData<List<Barcode>> {
        return mRepository.getSpecificBarcodes(barcode)
    }

    fun insert(barcode: Barcode) {
        mRepository.insert(barcode)
    }

    fun delete(barcode: Barcode) {
        mRepository.delete(barcode)
    }

    fun deleteAll() {
        mRepository.deleteAll()
    }

    fun update(barcode: Barcode) {
        mRepository.update(barcode)
    }
}
